src/videojs-http-streaming.js

```diff
@@ -384,6 +384,9 @@ class HlsHandler extends Component {
       this.tech_.setCurrentTime(time);
       this.setCurrentTime(time);
     };
+    if (this.source_.smoothQualityChange !== undefined) {
+      this.options_.smoothQualityChange = this.source_.smoothQualityChange;
```

Comment:
is this supposed to also be a source level option, or just a player level option?
If source level option is also valid, you should add `smoothQualityChange` to [hereinstead](https://github.com/videojs/http-streaming/blob/master/src/videojs-http-streaming.js#L356)

Reply:
Ah yes thanks, in an older version I was setting the quality change function hereish which required the MPC, and thus couldn't go in `setOptions`. But it can now, great catch.